src/map/Pins.js

```diff
@@ -104,6 +110,16 @@
                 }
             })
         ;
+        gPinEnter
+            .each(function(d){
+                if(d.ext && d.ext.tooltip){
```

Comment:
While unlikely a data row may start off with no tooltip data, but may get some later on - so you should add the event to every pin regardless.  I may mean we need some more logic elsewhere.

Reply:
I believe the pin data points are considered "new" if any aspect of them changes (line ~96 in Pins.js), so "enter" status should apply to the pins after a tooltip is added.

Reply:
Would this call to **.data()** key the data rows on index? or on the row's content?
```javascript
this.pinsPaths = this._pinsTransform.selectAll(".pin").data(this.visible() ? this.pinsData() : []);
```

Reply:
When no "id" function is passed into the ```data``` function on line ~96 it uses the idx position of the data row as its identifier.